### DISCUSSION TITLE
solution suddenly cannot be built and run

### DISCUSSION THREAD

**Discussion by mrchanshares** [File-New-Project/EarTrumpet#1124] [Q&A]
I have been working with my fork copy at 
https://github.com/mrchanshares/EarTrumpet with visual studio 2017 on window 10 .
And I have been working with the codes and debugging. Suddenly, my copy can no longer builds and runs. I get the following errors. Can you suggest a fix?

Error	MSB4057	The target "GetVersion" does not exist in the project.	C:\Users\a\Source\Repos\mrchanshares\EarTrumpet\EarTrumpet\EarTrumpet.csproj	

 "Build failed. Do you want to continue debugging?"

System.IO.IOException: 'Cannot locate resource 'app.xami'.'

An unhandled exception of type 'System.IO.IOException' occurred in PresentationFramework.dll
Cannot locate resource 'app.xaml'.

The final changes I made and submitted is on the file VolumeSlider.cs .

Also, if there are any bug in visual studio, it may have been fixed in latest release. Will the project be moved to latest release, such as visual studio 2022?

Thanks

**Comment by mrchanshares** [comment]
For unknown reason, the project can now run. Look like such debugging operation may be dependent on something external like some microsoft symbol servers etc. May be these are down yesterday. Just a guess.

The project still cannot be built due to followings:
C:\Users\a\Source\Repos\mrchanshares\EarTrumpet\EarTrumpet\EarTrumpet.csproj(501,30): error MSB4057: The target "GetVersion" does not exist in the project.

**Reply by riverar** [reply]
Which Visual Studio are you using? 2019? Did you maybe upgrade the GitVersion nuget packages by mistake?

**Reply by mrchanshares** [reply]
Now I can run the debug with Visual studio 2017 on window 10 and also Visual studio 2022 on window 11. It cannot build, however.  It displays "Build failed. Do you want to continue debugging?" 
As I am completely new to Visual studio. I don't know if a build is necessary or not too so long as it can run. I just copy and click the exe file in the debug folder and it can also run.
Also, I want to know if there is a UI design interface for the VolumeSlider control so I can look at it and hope to change the thickness of the two displayed volume bar for the left and right channels.
And, the mixerWindow  in one of the two computers occupy a big part of the screen. So is there way to change its size/resolution.

**Reply by mrchanshares** [reply]
Good news is that after being warned for "The breakpoint will not currently be hit." , I have changed the setting "To allow this for all breakpoints, disable the option 'Require source files to be exactly match the original version' under Tools, Options, Debugging, General". The project can now be built in Visual studio 2022 on window 11:

1>------ Rebuild started: Project: EarTrumpet, Configuration: Debug|x86 ------
1>C:\Users\lib20\Source\Repos\EarTrumpet\packages\GitVersionTask.5.5.1\build\GitVersionTask.targets(19,9): warning : File contains assembly version attributes which conflict with the attributes generated by GitVersion obj\x86\Debug\AssemblyInfo.g.cs
1>C:\Users\lib20\Source\Repos\EarTrumpet\EarTrumpet\ALINK : warning AL1073: Referenced assembly 'mscorlib.dll' targets a different processor

1>  EarTrumpet -> C:\Users\lib20\Source\Repos\EarTrumpet\Build\Debug\EarTrumpet.exe
========== Rebuild: 1 succeeded, 0 failed, 0 up-to-date, 0 skipped ==========
Rebuild succeeded.
